wgpu-hal/src/gles/adapter.rs

```diff
@@ -450,6 +450,12 @@ impl super::Adapter {
             supported((3, 1), (4, 2)) || extensions.contains("GL_ARB_shader_image_load_store"),
         );
         features.set(wgt::Features::SHADER_UNUSED_VERTEX_OUTPUT, true);
+        let query_buffers = extensions.contains("GL_ARB_query_buffer_object")
+            || extensions.contains("GL_AMD_query_buffer_object");
+        if extensions.contains("GL_ARB_timer_query") && query_buffers {
```

Comment:
```suggestion
        if extensions.contains("GL_ARB_timer_query") {
```

As we have the infrastructure to resolve without query_buffers, we should expose it.

We should however expose the blocking-ness to the user, as the user potentially can work around it. If we add a downlevel flag with a name like `NONBLOCKING_QUERY_RESOLVE` with a description like:

```rust
/// If this is true, calls to `CommandEncoder::resolve_query_set` will be performed on the queue timeline.
///
/// If this is false, calls to `CommandEncoder::resolve_query_set` will be performed on the device (i.e. cpu) timeline
/// and will block that timeline until the query has data. You may work around this limitation by waiting until the submit
/// whose queries you are resolving is fully finished (through use of `queue.on_submitted_work_done`) and only
/// then submitting the resolve_query_set command. The queries will be guarenteed finished, so will not block.
///
/// Supported by:
/// - Vulkan,
/// - DX12
/// - Metal
/// - OpenGL 4.2+
///
/// Not Supported by:
/// - GL ES / WebGL
```